### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by DasSkelett**
## Some dependency updates
* NuGet 5.0.2 to 5.3.1
* Newtonsoft.Json 12.0.2 to 12.0.3
* Autofac 4.9.2 to 4.9.4
* AWSSDK.Core 3.3.103 to 3.3.103.64
* AWSSDK.SQS 3.3.100.34 to 3.3.102.31
* NJsonSchema 10.0.19 to 10.0.27
* Namotion.reflection 1.0.5 to 1.0.7
* Moq 4.11.0 to 4.13.1
* NUnit3TestAdapter 3.13.0 to 3.15.1
* System.Runtime.CompilerServices.Unsafe 4.5.2 to 4.6.0
* System.Threading.Tasks.Extensions 4.5.2 to 4.5.3

The update of NuGet to 5.3.1 should solve the problems of #2842 (and if we're lucky #2898 too).
If this is right, the msbuild dependency of the AUR PKGBUILD can be removed again.

## BOMs removed
I removed some Byte Order Marks and sorted the using directives in NetKAN a bit.
I followed this rule that was already used in most cases:
First System.* imports
Then other external imports (log4net, Autofac...)
Then CKAN.\* (where \* != NetKAN.*) imports
Then CKAN.NetKAN.\* imports

In each group they are sorted alphabetically

### CODE REVIEW

**Comment by DasSkelett** [comment]
@politas can you remove the `msbuild` dependency from the AUR package's PKGBUILD and try to build CKAN?

**Comment by DasSkelett** [comment]
The new commit adds descriptions to all the Cake targets, check them out with `./build --showdescriptions` or `./build -s`:
```
Task                               Description
=====================================================================================
Default                            Build ckan.exe and netkan.exe, if not specified otherwise in Debug configuration for .NET Framework.
Debug                              Build ckan.exe and netkan.exe in Debug configuration, if not specified otherwise for .NET Framework.
Release                            Build ckan.exe and netkan.exe in Release configuration, if not specified otherwise for .NET Framework.
Ckan                               Build only ckan.exe, if not specified otherwise in Debug configuration for .NET Framework.
Netkan                             Build only netkan.exe, if not specified otherwise in Debug configuration for .NET Framework.
DLL                                Build only ckan.dll (CKAN-Core) for .NET Core.
docker-inflator                    Build the Docker image for the Inflator and push it to Dockerhub.
osx                                Build the macOS(OSX) dmg package.
osx-clean                          Clean the output directory of the macOS(OSX) package.
deb                                Build the deb package for Debian-absed distros.
deb-test                           Test the deb packaging.
deb-clean                          Clean the deb output directory.
rpm                                Build the rpm package for RPM-based distros.
rpm-test                           Test the rpm packaging.
rpm-clean                          Clean the rpm package output directory.
Restore-Nuget                      Intermediate - Download dependencies with NuGet when building for .NET Framework.
Build-DotNet                       Intermediate - Call MSBuild/XBuild to build the CKAN.sln.
Restore-DotNetCore                 Intermediate - Download dependencies with NuGet when building for .NET Core.
Build-DotNetCore                   Intermediate - Call .NET Core's MSBuild to build the ckan.dll.
Generate-GlobalAssemblyVersionInfo Intermediate - Calculate the version strings for the assembly.
Repack-Ckan                        Intermediate - Merge all the separate DLLs and EXEs to a single executable.
Repack-Netkan                      Intermediate - Merge all the separate DLLs and EXEs to a single executable.
Test                               Run CKANs tests after compilation.
Test+Only                          Only run CKANs tests, without compiling beforehand.
Test-UnitTests+Only                Intermediate - Only run CKANs unit tests, without compiling beforehand.
Test-UnitTests+Only-DotNetCore     Intermediate - Only run CKANs unit tests using DotNetCoreTest, without compiling beforehand.
Test-Executables+Only              Intermediate - Only test CKANs executables, without compiling them beforhand.
Test-CkanExecutable+Only           Intermediate - Only test the ckan.exe, without compiling beforhand.
Test-NetkanExecutable+Only         Intermediate - Only test the netkan.exe, without compiling beforhand.
Version                            Print the current CKAN version.
```

I also added a new Cake target called `DLL`, it only runs `Build-DotNetCore` and is a shortcut for `./build --configuration=*_NetCore`.

Because Cake calculates whether to execute a target or not based on the `WithCriteria` modifier on load time by default, I changed them to lambdas. This allows changing `buildFramework` in `Setup()`.

File: build.cake [hunk 1]
```diff
@@ -82,44 +93,53 @@ Task("docker-inflator")
...
     .Does(() => StartProcess("make",
         new ProcessSettings { WorkingDirectory = "macosx" }));
 
 Task("osx-clean")
+    .Description("Clean the output directory of the macOS(OSX) package.")
     .Does(() => StartProcess("make",
         new ProcessSettings { Arguments = "clean", WorkingDirectory = "macosx" }));
 
 Task("deb")
+    .Description("Build the deb package for Debian-absed distros.")
```
Comment (HebaruSan) [inline build.cake]: Minor typo, absed -> based

Comment (DasSkelett) [reply build.cake]: Debian derivatives are very muscular :man_shrugging:

**Pull request merged**